56.md

```diff
@@ -0,0 +1,70 @@
+
+NIP-56
+======
+
+Reporting
+---------
+
+`draft` `optional` `author:jb55`
+
+A report is a `kind 1984` note that is used to report other notes for spam,
+illegal and explicit content.
+
+The content MAY contain additional information submitted by the entity
+reporting the content.
+
+Tags
+----
+
+The report event MUST include a `p` tag referencing the pubkey of the user you
+are reporting.
+
+If reporting a note, an `e` tag MUST also be included referencing the note id.
+
+A `report` tag MUST be included which consists of the following report types:
+
+- `explicit` - explicit content which may not be appropriate for minors: nudity, profanity, etc
+- `illegal` - something which may be illegal in some jurisdiction
+- `spam` - spam
+- `impersonation` - someone pretending to be someone else
+
+Some report tags only make sense for profile reports, such as `impersonation`
+
+Example code
+
+```swift
+enum ReportType: String {
+    case explicit
+    case illegal
+    case spam
+    case impersonation
+}
+
+func create_report_tags(target: ReportTarget, type: ReportType) -> [[String]] {
+    var tags: [[String]]
+    switch target {
+    case .user(let pubkey):
+        tags = [["p", pubkey]]
+    case .note(let notet):
+        tags = [["e", notet.note_id], ["p", notet.pubkey]]
+    }
+
+    tags.append(["report", type.rawValue])
+    return tags
+}
+```
```

Comment:
How about actual example events instead of the example code? The code is a

````suggestion
Example events
--------------

```json
{
  "kind": 1984,
  "tags": [
    [ "report", "explicit"],
    [ "p", <pubkey>]
  ],
  "content": "",
  ...
}

{
  "kind": 1984,
  "tags": [
    [ "report", "illegal"],
    [ "e", <eventId>],
    [ "p", <pubkey>]
  ],
  "content": "He's insulting the king!",
  ...
}

{
  "kind": 1984,
  "tags": [
    [ "report", "impersonation"],
    [ "p", <pubkey>]
  ],
  "content": "Profile is imitating npub12345...",
  ...
}
```
````

Reply:
Writing these examples I noticed it would be helpful to better reference impersonation victims with their own p-tag, so maybe the target of the report could carry the report-tag value?

```json
{
  "kind": 1984,
  "tags": [
    [ "p", <pubkey>, "explicit"]
  ],
  "content": "",
  ...
}

{
  "kind": 1984,
  "tags": [
    [ "e", <eventId>, "illegal"],
    [ "p", <pubkey>]
  ],
  "content": "He's insulting the king!",
  ...
}

{
  "kind": 1984,
  "tags": [
    [ "p", <impersonator pubkey>, "impersonation"],
    [ "p", <victim pubkey>]
  ],
  "content": "Profile is imitating #[2]",
  ...
}
```

Reply:
I like that better

Reply:
I mean the report kinds in the tags themselves.  But I like showing the events too.

Reply:
yeah this is fine I suppose, report kinds in the tags without a separate report tag